db/migrate/20241229165332_add_devise_to_users.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[8.0]
+  def change
+    rename_column :users, :crypted, :legacy_password_hash
+    change_column_null :users, :legacy_password_hash, true
+
+    change_table :users do |t|
+      ## Database authenticatable
+      # t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
```

Comment:
We don't need all of these extra modules (esp trackable, lockable), but I think it's worth having this information for audit purposes downstream, or improving security later on.